include/ginkgo/core/matrix/csr.hpp

```diff
@@ -981,6 +985,30 @@ class Csr : public EnableLinOp<Csr<ValueType, IndexType>>,
           strategy_(strategy->copy())
     {}
 
+    /**
+     * Creates a CSR matrix from already allocated (and initialized) row
+     * pointer array.
+     *
+     * @param exec  Executor associated to the matrix
+     * @param size  size of the matrix
+     * @param row_ptrs  array of row pointers
+     */
+    explicit Csr(
+        std::shared_ptr<const Executor> exec, const dim<2>& size,
+        Array<IndexType>&& row_ptrs,
+        std::shared_ptr<strategy_type> strategy = std::make_shared<sparselib>())
+        : EnableLinOp<Csr>(exec, size),
+          row_ptrs_{exec, std::move(row_ptrs)},
+          srow_(exec),
+          strategy_(strategy->copy())
+    {
+        GKO_ASSERT(row_ptrs_.get_num_elems() == size[0] + 1);
+        auto num_nnz = exec->copy_val_to_host(row_ptrs_.get_data() + size[0]);
+        values_ = Array<ValueType>(exec, num_nnz);
+        col_idxs_ = Array<IndexType>(exec, num_nnz);
+        this->make_srow();
+    }
```

Comment:
I don't think this is a common enough operation that we need a separate constructor for it.
The same can be accomplished with `create(exec, size, row_ptrs, Array<ValueType>(exec, nnz), Array<IndexType>(exec, nnz), strategy)`.
More in detail: For each matrix type, we have a set of size-based constructor and a single constructor from existing data. I think we should leave it at that separation, to create a uniform interface.